cmd/juju/application/upgradecharm.go

```diff
@@ -492,10 +501,12 @@ func (c *upgradeCharmCommand) addCharm(
 	config *config.Config,
 	oldURL *charm.URL,
 	charmRef string,
+	deployedSeries string,
 ) (charmstore.CharmID, *macaroon.Macaroon, error) {
 	var id charmstore.CharmID
-	// Charm may have been supplied via a path reference.
-	ch, newURL, err := charmrepo.NewCharmAtPathForceSeries(charmRef, oldURL.Series, c.ForceSeries)
+	// Charm may have been supplied via a path reference. If so, build a
+	// local charm URL from the deployed series.
+	ch, newURL, err := charmrepo.NewCharmAtPathForceSeries(charmRef, deployedSeries, c.ForceSeries)
```

Comment:
There are three more references to olduURL.Series below in this method. Can you update them to deployedSeries please?